pydantic/schema.py

```diff
@@ -831,7 +833,14 @@ def go(type_: Any) -> Type[Any]:
         attrs: Optional[Tuple[str, ...]] = None
         constraint_func: Optional[Callable[..., type]] = None
         if isinstance(type_, type):
-            if issubclass(type_, str) and not issubclass(type_, (EmailStr, AnyUrl, ConstrainedStr)):
+            if issubclass(type_, (SecretStr, SecretBytes)):
+                attrs = ('max_length', 'min_length')
+
+                def _secret_constraint(**kwargs: Any) -> Type[Any]:
```

Comment:
can't you just do
```suggestion
                def constraint_func(**kwargs: Any) -> Type[Any]:
```
and remove `constraint_func = _secret_constraint` below?